### DISCUSSION TITLE
Offline installation of Robots for Rhino 7

### DISCUSSION THREAD

**Discussion by Paultn12** [visose/Robots#217] [Q&A] [answered]
Hi!

We are currently trying to battle our University security to get Robots installed on University computers; they do not currently allow Rhino to access the internet...

We have a workshop on Robotic Construction this coming Wednesday and looking for a plan B to have The Robots plugin available.

Would anyone be able to share the offline files we need to install Robots on Rhino 7? Our robot should already be in the drop down. Bringing the files over from an installed laptop wasn't successful.

Thank You in advance!
Paul.

**Comment by visose** [comment, marked as answer]
Hi @Paultn12, the proper solution is to convince IT to allow Rhino to access the internet. Today most software will be unusable or at least handicapped without access to the internet.
For Robots you'll need internet for 2 things: Installing the plugin and downloading robot libraries. You might be able to get around this with some work arounds:

### For robot libraries
You can download the .xml/.3dm pair of files from the `libraries` branch and copy them to the `Documents\Robots` folder.

### To install the plugin
1. Download the latest .yak package from here: https://files.mcneel.com/yak/packages/Robots-1.6.4-rh7_0-any.yak
This link is for the current latest version. If there's a new release you should edit the link accordingly.
2. If the lab computers are on a network and all have access to a shared network folder, copy the yak file this shared folder.
3. On each Rhino, go to Options > Advanced and add the path to this shared folder as a package manager source:
<img width="568" alt="image" src="https://user-images.githubusercontent.com/5129614/219975536-35124662-d8a9-4e4b-8593-07ad42f69a2f.png">
4. Restart Rhino, open the PackageManager and Robots should show up. Install as usual.

If there's no access to a shared folder, you'll have to copy the .yak file to each computer and as source use a path to a local folder.

**Reply by Paultn12** [reply]
Thank you Vincente,

and firstly, I couldn't agree more. Hopefully they will come around and allow access.

Thank you for the walkthrough, but I am not having much luck with what you suggested. I am going for option 4 because I don't have admin for the room I am trying to install, but I am getting this issue still.

![image](https://user-images.githubusercontent.com/25488330/220138610-a748800a-895a-48be-966a-9f233c6313da.png)

Thank you again for your help.
Paul.

**Reply by Paultn12** [reply]
Ah, but I have just looked at my screenshot... I used a path to the downloads to replace the default.

C:\Users\pn****ls\Downloads\Robots-1.6.4-rh7_0-any.yak

**Reply by visose** [reply]
If the yak file is in `C:\Some\Folder\Robots-1.6.4-rh7_0-any.yak`, then the Sources value would be: `C:\Some\Folder\`

**Reply by visose** [reply]
Btw, the numbered bullet points in my message above are steps, not options. The last sentence is a comment separate from the steps.

**Reply by Paultn12** [reply]
Hi Vincente,

Thank you for your help, that has worked perfectly.

Paul.
